packages/core-compat-api/src/collectLegacyRoutes.tsx

```diff
@@ -186,7 +185,12 @@ export function collectLegacyRoutes(
         routeElement,
         'core.mountPoint',
       );
-      if (!plugin || !path) {
+      if (path === undefined) {
+        throw new Error(
+          `<Route /> element with invalid path has been detected. Please make sure to pass the path's props`,
+        );
+      }
+      if (!plugin) {
         return;
```

Comment:
> [!NOTE]
I haven't thrown anything in case no plugin is detected to avoid getting troubles with react router's corner cases such as `<Navigate />`

Reply:
I think we should throw an error in this case too, and let users know that they can remove the route if needed. Something like `Route with path '...' has en element that can not be converted as it does not belong to a plugin. Make sure that the top-level React element of the element prop is an extension from a Backstage plugin, or remove the Route completely. See <link-to-app-migration-docs> for more info`. In the migration docs we can then mention that you should remove the `Navigation` element and replace it with configuration of the `app/routes` extension (which we should also add).